presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -185,9 +187,16 @@ public StreamProperties visitJoin(JoinNode node, List<StreamProperties> inputPro
                             .translate(column -> PropertyDerivations.filterOrRewrite(outputs, node.getCriteria(), column))
                             .unordered(unordered);
                 case LEFT:
-                    return leftProperties
-                            .translate(column -> PropertyDerivations.filterIfMissing(outputs, column))
-                            .unordered(unordered);
+                    if (nativeExecution && node.getCriteria().isEmpty()) {
+                        // This maps to a NestedLoopJoin in Native engine. The NestedLoopJoin output is not
```

Comment:
@rschlussel @feilong-liu Rebecca, Feilong, would you help take a first pass for this change? At a high level this seems reasonable, but I haven't been working in the optimizer code for quite some time.

Reply:
does only left join with no criteria produce nested loop join?  What happens for inner join?  Also, would it be enough to just change unordered to false for these cases, but otherwise keep the other things the same (does the stream distribution or partitioning change)?

Reply:
@rschlussel : That's a good point. I was trying to contain this fix, but this change would be applicable for inner too. This change has come by in correlated queries that seem to use LEFT join predominantly.

But just changing unordered to false didn't work. I had to change the partitioning as well for the change to take effect. The addLocalExchanges seems to leverage distribution as well. More at https://github.com/prestodb/presto/issues/22585#issuecomment-2168942249